File: Onoicrm.Front/src/modules/booking/entities/Booking.ts
import {Entity} from "~/shared/entities/Entity";
import {UserRegisterModel} from "~/modules/user/models/UserResisterModel";
export enum DiscountType {Percent, Money}
export enum SalaryType {Salary, Percent, ServicePercent }
export class Booking extends Entity {
  public doctorId: number| null = null;
  public patientId: number| null = null;
  public doctor:any;
  public serviceGroupId:any
  public armchairId: number | null = null;
  public clinicId: number| null = null;
  public description: string = "";
  public сomplaint: string = "";
  public dateTimeStart:string = "";
  public dateTimeEnd:string = "";
  public patient?: any;
  public bookingServices:any[] = []
  public serviceGroups:any[] = []
  public bookingTeeth:any[] = []
  public discount:number = 0;
  public discountType:DiscountType = DiscountType.Percent;
  public implementedServices:any[] = []
  public salary:number = 0;
  public salaryType:SalaryType = SalaryType.Percent;
  constructor(instance?: Partial<Booking>) {
    super();
    Object.assign(this, instance);
  }
}


export class BookingModel {
  public booking!:Booking;
  public serviceGroupId?:number|null = null
  constructor(instance?: Partial<BookingModel>) {
    Object.assign(this, instance);
  }
}


export enum PatientTypeId {
  Exist,
  New
}

File: Onoicrm.Domain/Entities/Booking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Globalization;

namespace Onoicrm.Domain.Entities;

public enum DiscountType {Percent, Money}

public class Booking : Entity
{    
    public string Complaint { get; set; } = "";
    public string Caption { get; set; } = "";
    public string Description { get; set; } = "";
    public DateTime DateTimeStart { get; set; }
    public DateTime DateTimeEnd { get; set; }
    [Required]
    public long? DoctorId { get; set; }
    public UserProfile? Doctor { get; set; }
    public long? PatientId { get; set; }
    public Patient? Patient { get; set; }
    public long? ArmchairId { get; set; }
    public Armchair? Armchair { get; set; }
    [Required]
    public long? ClinicId { get; set; }
    public Clinic? Clinic { get; set; }

    public long Salary { get; set; }
    public SalaryType SalaryType { get; set; } = SalaryType.Percent;
    
    public long? TreatmentPlanId { get; set; }
    public TreatmentPlan? TreatmentPlan { get; set; }
    public long Discount { get; set; }
    public DiscountType DiscountType { get; set; }
    public long? BookingGroupId { get; set; }
    public BookingGroup? BookingGroup { get; set; }
    
    public long? ServiceGroupId { get; set; }
    public ServiceGroup? ServiceGroup { get; set; }
    public ICollection<BookingFile> Files { get; set; } = new List<BookingFile>();
    public ICollection<BookingTooth> BookingTeeth { get; set; } = new List<BookingTooth>();
    public ICollection<ImplementedService> ImplementedServices { get; set; } = new List<ImplementedService>();
    [NotMapped]
    public string DateTime
    {
        get
        {
           var formattedDateTimeStart = DateTimeStart.AddHours(6).ToString("dd-MMMM HH:mm", CultureInfo.GetCultureInfo("ru-RU"));
           var formattedDateTimeEnd = DateTimeEnd.AddHours(6).ToString("HH:mm", CultureInfo.GetCultureInfo("ru-RU"));
           return $"{formattedDateTimeStart}-{formattedDateTimeEnd}"; 
        }
    }
    
    public decimal GetTotal(decimal sum) => DiscountType switch
    {
        DiscountType.Money => sum - Discount,
        DiscountType.Percent => Discount > 0 ? sum - sum * Discount / 100 : sum,
        _ => throw new ArgumentOutOfRangeException()
    };

    protected override int GetClassId() =>  ClassNames.Booking;
}